src/main/java/io/vertx/core/impl/DeploymentManager.java

```diff
@@ -374,19 +376,68 @@ public void unregisterVerticleFactory(VerticleFactory factory) {
     return factoryList;
   }
 
+  private static URL mapToURL(String path) {
+    try {
+      File file = new File(path);
+      return file.toURI().toURL();
+    } catch (MalformedURLException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  private static List<URL> extractCPFromProperty() {
+    String classpath = System.getProperty("java.class.path");
+    List<URL> classpathURLs = new ArrayList<>();
+    for (String path : classpath.split(":")) {
+      classpathURLs.add(mapToURL(path));
+    }
+    return classpathURLs;
+  }
+
+  private static List<URL> extractCPByManifest() {
+    List<URL> classpathURLs = new ArrayList<>();
+    String searchFile = "META-INF/MANIFEST.MF";
+    String suffix = "!/" + searchFile;
+    String prefix = "file:";
+    Enumeration<URL> urls;
+    try {
+      urls = DeploymentManager.class.getClassLoader().getResources(searchFile);
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
+    }
+    for (URL url : Collections.list(urls)) {
+      String path = url.getFile().replace(prefix, "").replace(suffix, "").trim();
```

Comment:
This is specific, typically in OSGi there are not `file:`.

Reply:
But if there is not "file:", it will just not replace it. Or do you mean, there could be something else and  I should delete everything from beginning to the first colon?

Reply:
I now fixed it, but not sure if this is a proper solution.